scheduler/pkg/klcpermit/workflow_manager.go

```diff
@@ -122,14 +125,59 @@ func (sMgr *WorkloadManager) getSpan(ctx context.Context, crd *unstructured.Unst
 	return ctx, span
 }
 
+// CreateResourceName is a function that concatenates the parts from the
+// input and checks, if the resulting string matches the maxLen condition.
+// If it does not match, it reduces the subparts, starting with the first
+// one (but leaving its length at least in minSubstrLen so it's not deleted
+// completely) adn continuing with the rest if needed.
+// Let's take WOrkloadInstance as an example (3 parts: app, workload, version).
+// First the app anme is reduced if needed (only to minSubstrLen),
+// afterwards workload and the version is not reduced at all. This pattern is
+// chosen to not reduce only one part of the name (that can be completely gona
+// afterwards), but to include all of the parts in the resulting string.
+func createResourceName(maxLen int, minSubstrLen int, str ...string) string {
```

Comment:
fine for now to have this duplication, but in the future we should definitely look into go workspaces to move such things into a common module :) I have a created a ticket for that: #1476

Reply:
completely agree, we have also WI definitions in scheduler duplicated, what is a bigger problem than one function